tests/src/tests/mixnet.rs

```diff
@@ -0,0 +1,106 @@
+use std::{
+    collections::HashMap,
+    net::{Ipv4Addr, SocketAddr, SocketAddrV4},
+};
+
+use mixnet_client::{MixnetClient, MixnetClientConfig};
+use mixnet_node::{MixnetNode, MixnetNodeConfig};
+use mixnet_topology::{Layer, MixnetTopology, Node};
+use rand::{rngs::OsRng, RngCore};
+
+#[tokio::test]
+async fn mixnet() {
+    let topology = run_mixnodes();
+    let (client1, client2, destination) = run_clients(topology.clone()).await;
+
+    let mut msg = [0u8; 100 * 1024];
+    rand::thread_rng().fill_bytes(&mut msg);
+
+    let res = client1.send(msg.to_vec(), destination, &mut OsRng, topology.layers.len());
```

Comment:
I just changed the mixnet design a bit in the PR #307, so that the message sender doesn't need to specify the mixnet destination explicitly. Please review that PR together with this PR. I put detailed descriptions there.